File: lc/c/070.c
int search(int* nums, int numsSize, int target){
    int l = 0, r = numsSize-1;
    while (l <= r) {
        int m = l + (r-l)/2;
        if (nums[m] < target)
            l = m+1;
        else if (nums[m] > target)
            r = m-1;
        else
            return m;
    }
    return -1;
}

File: lc/py/070.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums)-1
        while l <= r:
            m = l + (r-l)//2
            if nums[m] < target:
                l = m+1
            elif nums[m] > target:
                r = m-1
            else:
                return m
        return -1

# BISECT
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i = bisect.bisect_left(nums, target)
        return i if  i < len(nums) and nums[i] == target else -1
